scripts/create_testenv.sh

```diff
@@ -101,9 +101,3 @@ pip install --no-cache-dir -r requirements-dev.txt
 pip install --no-cache-dir -r requirements-docs.txt
 pip install --no-cache-dir -r requirements-external.txt
 pip install --no-cache-dir -r requirements-optional.txt
-
-
-conda install -y geckodriver firefox jupyterlab ipywidgets nodejs --channel conda-forge
```

Comment:
This may be fine but we may need to delete the notebook portion of the container script since it will break. Can you verify that either the the notebook launcher works or remove it entirely?

https://github.com/arviz-devs/arviz/blob/main/scripts/container.sh#L49

Reply:
I think we should keep the jupyterlab and jupyternotebook options in docker which means fixing that. We needed geckodriver and firefox for doc building with bokeh png previews, so these two can be deleted and should be deleted.

Can you try if removing these fixed the issue already or if more work is needed? (in addition to as Ravin suggested making sure that the noetbook and lab flags continue to work)

Reply:
Sure, it took me a whole day to set up the project, so I will try it close to the weekend, so If I encounter any issues,  I'll fix them on weekend.

Reply:
I tried removing only geckodriver and firefox, keeping only`pymc3` instead of `pymc3 @ git+https://github.com/pymc-devs/pymc3` and it failed. 

```
#13 6781.5 ERROR: Exception:
#13 6781.5 Traceback (most recent call last):
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_vendor/urllib3/response.py", line 438, in _error_catcher
#13 6781.5     yield
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_vendor/urllib3/response.py", line 519, in read
#13 6781.5     data = self._fp.read(amt) if not fp_closed else b""
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/http/client.py", line 459, in read
#13 6781.5     n = self.readinto(b)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/http/client.py", line 503, in readinto
#13 6781.5     n = self.fp.readinto(b)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/socket.py", line 669, in readinto
#13 6781.5     return self._sock.recv_into(b)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/ssl.py", line 1241, in recv_into
#13 6781.5     return self.read(nbytes, buffer)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/ssl.py", line 1099, in read
#13 6781.5     return self._sslobj.read(len, buffer)
#13 6781.5 socket.timeout: The read operation timed out
#13 6781.5 
#13 6781.5 During handling of the above exception, another exception occurred:
#13 6781.5 
#13 6781.5 Traceback (most recent call last):
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/cli/base_command.py", line 173, in _main
#13 6781.5     status = self.run(options, args)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/cli/req_command.py", line 203, in wrapper
#13 6781.5     return func(self, options, args)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/commands/install.py", line 315, in run
#13 6781.5     requirement_set = resolver.resolve(
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/resolver.py", line 94, in resolve
#13 6781.5     result = self._result = resolver.resolve(
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_vendor/resolvelib/resolvers.py", line 472, in resolve
#13 6781.5     state = resolution.resolve(requirements, max_rounds=max_rounds)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_vendor/resolvelib/resolvers.py", line 366, in resolve
#13 6781.5     failure_causes = self._attempt_to_pin_criterion(name)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_vendor/resolvelib/resolvers.py", line 212, in _attempt_to_pin_criterion
#13 6781.5     criteria = self._get_updated_criteria(candidate)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_vendor/resolvelib/resolvers.py", line 203, in _get_updated_criteria
#13 6781.5     self._add_to_criteria(criteria, requirement, parent=candidate)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_vendor/resolvelib/resolvers.py", line 172, in _add_to_criteria
#13 6781.5     if not criterion.candidates:
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_vendor/resolvelib/structs.py", line 151, in __bool__
#13 6781.5     return bool(self._sequence)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/found_candidates.py", line 140, in __bool__
#13 6781.5     return any(self)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/found_candidates.py", line 128, in <genexpr>
#13 6781.5     return (c for c in iterator if id(c) not in self._incompatible_ids)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/found_candidates.py", line 32, in _iter_built
#13 6781.5     candidate = func()
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/factory.py", line 204, in _make_candidate_from_link
#13 6781.5     self._link_candidate_cache[link] = LinkCandidate(
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 295, in __init__
#13 6781.5     super().__init__(
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 156, in __init__
#13 6781.5     self.dist = self._prepare()
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 227, in _prepare
#13 6781.5     dist = self._prepare_distribution()
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/resolution/resolvelib/candidates.py", line 305, in _prepare_distribution
#13 6781.5     return self._factory.preparer.prepare_linked_requirement(
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/operations/prepare.py", line 508, in prepare_linked_requirement
#13 6781.5     return self._prepare_linked_requirement(req, parallel_builds)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/operations/prepare.py", line 550, in _prepare_linked_requirement
#13 6781.5     local_file = unpack_url(
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/operations/prepare.py", line 239, in unpack_url
#13 6781.5     file = get_http_url(
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/operations/prepare.py", line 102, in get_http_url
#13 6781.5     from_path, content_type = download(link, temp_dir.path)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/network/download.py", line 145, in __call__
#13 6781.5     for chunk in chunks:
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/cli/progress_bars.py", line 144, in iter
#13 6781.5     for x in it:
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_internal/network/utils.py", line 63, in response_chunks
#13 6781.5     for chunk in response.raw.stream(
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_vendor/urllib3/response.py", line 576, in stream
#13 6781.5     data = self.read(amt=amt, decode_content=decode_content)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_vendor/urllib3/response.py", line 541, in read
#13 6781.5     raise IncompleteRead(self._fp_bytes_read, self.length_remaining)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/contextlib.py", line 131, in __exit__
#13 6781.5     self.gen.throw(type, value, traceback)
#13 6781.5   File "/usr/local/envs/testenv_3.8_PYSTAN_latest_PYRO_latest_EMCEE_latest_TF_latest/lib/python3.8/site-packages/pip/_vendor/urllib3/response.py", line 443, in _error_catcher
#13 6781.5     raise ReadTimeoutError(self._pool, None, "Read timed out.")
#13 6781.5 pip._vendor.urllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='files.pythonhosted.org', port=443): Read timed out.
------
executor failed running [/bin/sh -c ./scripts/create_testenv.sh]: exit code: 2
```

Reply:
Sorry for the delay in response. These timeouts are related to internet issues, not package conflicts. Ill try and figure out what the right thing to do here is

Reply:
Actually, I discussed it with @OriolAbril, he has asked me to try one or two things, which I'll soon and push changes if they get successful.